src/test/kotlin/espressoLogcat/BaseTests.kt

```diff
@@ -7,13 +7,15 @@ import java.io.File
 open class BaseTests {
     @Before
     fun init() {
+        dataList = getFileAsMutableList(filename = testDataFilename)
+
         lineDataMap[lineDataEventOne.date] = lineDataEventOne
         lineDataMap[lineDataEventTwo.date] = lineDataEventTwo
         lineDataMap[lineDataEventThree.date] = lineDataEventThree
         lineDataMap[lineDataEventFour.date] = lineDataEventFour
     }
 
-    protected val mockLogcatExecutor = Mockito.mock(LogcatExecutor::class.java)
+    protected val mockLogcatExecutorImpl: LogcatExecutorImpl = Mockito.mock(LogcatExecutorImpl::class.java)
```

Comment:
Doesn't matter a whole ton but you should mock out the interface instead of the implementation. If you wanted to, you could even drop Mockito entirely for this and just make a test implementation subclass that extends LogcatExecutor and pulls from the the test resources

Reply:
I think I did this in e4b2712, lemme know if you were thinking something different.

Reply:
looks good